app/models/poll.rb

```diff
@@ -97,6 +70,25 @@ def date_should_be_in_future
   end
 
   def close_if_expire
-    closed! if expire_at&. < DateTime.now
+    finish! if expire_at&. < DateTime.now
+  end
+
+  def status_machine
+    @status_machine ||= configure_machine(status)
```

Comment:
Do you need this variable anywhere? If no, I'd make `configure_machine` implicit, putting it inside of some after-load callback, and set up that wiring inside of your wonderful `PollStatusMachine` module...